test/TestWeaselIPC/TestWeaselIPC.cpp

```diff
@@ -168,12 +168,12 @@ class TestRequestHandler : public weasel::RequestHandler
 		std::cerr << "RemoveClient: " << session_id << std::endl;
 		return 0;
 	}
-	virtual BOOL ProcessKeyEvent(weasel::KeyEvent keyEvent, UINT session_id, LPWSTR buffer) {
+	virtual BOOL ProcessKeyEvent(weasel::KeyEvent keyEvent, UINT session_id, EatLine eat) {
 		std::cerr << "ProcessKeyEvent: " << session_id 
 			  << " keycode: " << keyEvent.keycode 
 			  << " mask: " << keyEvent.mask 
 			  << std::endl;
-		wsprintf(buffer, L"Greeting=Hello, 小狼毫.\n");
+		eat(std::wstring(L"Greeting=Hello, 小狼毫.\n"));

```

Comment:
How does `TestWeaselIPC` client actually output in your build environment?
In my case, the greeting message gets truncated and the output looks like this:
```
server replies: 1
get response data: 1
buffer reads:
Greeting=Hello, ?请按任意键继续. . .
```

Reply:
Works fine in my terminal. Seems like a encoding issue.
![image](https://user-images.githubusercontent.com/10287709/36582274-cb9d8ae0-18ab-11e8-99fc-dc637964d82f.png)

Reply:
Curiously, the output is fine when redirected to file output. Maybe I should just assume it works all right.

Reply:
After change my CMD charset to UTF-8 it shows like this:
```text
λ  .\TestWeaselIPC.exe
server replies: 1
get response data: 1
buffer reads:
Greeting=Hello, ???.
```
I think it can only display correctly in GB2312?

Reply:
Well the problem cannot be in the charset if your system language is also zh-CN...
I've checked the memory buffer showing nothing wrong.